api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForLessThanXmonthsCalcultion.java

```diff
@@ -1,31 +1,37 @@
 package org.openmrs.module.eptsreports.reporting.calculation.pvls;
 
+import java.util.Collection;
+import java.util.Date;
+import java.util.Map;
+
 import org.openmrs.Concept;
 import org.openmrs.Obs;
-import org.openmrs.api.context.Context;
 import org.openmrs.calculation.patient.PatientCalculationContext;
 import org.openmrs.calculation.result.CalculationResultMap;
 import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
 import org.openmrs.module.eptsreports.reporting.calculation.AbstractPatientCalculation;
 import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
 import org.openmrs.module.eptsreports.reporting.calculation.EptsCalculations;
 import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
-import java.util.Collection;
-import java.util.Date;
-import java.util.Map;
-import java.util.Set;
-
+@Component
 public class OnArtForLessThanXmonthsCalcultion extends AbstractPatientCalculation {
 	
+	@Autowired
+	private HivMetadata hivMetadata;
+	
 	@Override
 	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> params, PatientCalculationContext context) {
 		
 		CalculationResultMap map = new CalculationResultMap();
-		Concept viralLoadConceptuuid = Context.getConceptService().getConceptByUuid("e1d6247e-1d5f-11e0-b929-000c29ad1d07");
+		Concept viralLoadConcept = hivMetadata.getHivViralLoadConcept();
+		
 		// get data inicio TARV
 		CalculationResultMap arvsInitiationDateMap = calculate(new InitialArtStartDateCalculation(), cohort, context);
-		CalculationResultMap lastVl = EptsCalculations.lastObs(viralLoadConceptuuid, cohort, context);
+		CalculationResultMap lastVl = EptsCalculations.lastObs(viralLoadConcept, cohort, context);
```

Comment:
I always get a null pointer exception when I Autowire components and use them, this made me revert to the current way am using. Did you test that and make sure the NPE is NOT being thrown?

Reply:
@ningosi , that's weird, but there is no NPE in that code, you can check. maybe you were not putting the correct annotation in the component class.

Reply:
Works for me too, I am not getting a NPE. Agree with Guimino's comment about the annotations and we've used this approach in the other reports so should work here too.